[Avinash P] iP

avinazz3 opened the pull request.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

avinazz3 committed:
Rename, Greet and Exit

avinazz3 committed:
Echo

avinazz3 committed:
Add, List

avinazz3 committed:
Mark as done

avinazz3 committed:
Todos, Events, Deadlines

avinazz3 committed:
Automated Text UI Testing

avinazz3 committed:
Handle Errors. Massive refactoring

avinazz3 committed:
Delete

avinazz3 committed:
Save

avinazz3 committed:
Date and Time

avinazz3 committed:
A-MoreOOP. The requirements of MoreOOP has already been fulfilled in earlier increments. this is an empty commit.

avinazz3 committed:
refactor(babe): reorganize packages and files

Refactored files and packages under the babe package.

avinazz3 committed:
Modified(build.gradle) to match rest of the project

avinazz3 committed:
Merge branch 'add-gradle-support'

avinazz3 committed:
test: add JUnit tests for Parser and Storage classes

Add JUnit tests to verify core functionality:
- Test Parser.createDeadline() for valid/invalid inputs
- Test Storage.save() and Storage.load() for multiple scenarios

avinazz3 committed:
build: package app as executable JAR

avinazz3 committed:
Add JavaDoc comments to classes and methods

- Added header and method-level JavaDoc comments for better documentation
- Covered most non-private classes and methods with detailed explanations
- Documented non-trivial private methods for clarity

avinazz3 committed:
Add task search functionality by keyword

- Add findTasks method to TaskList to search task descriptions
- Implement FindCommand to handle 'find' user input
- Support case-insensitive keyword matching in task descriptions
- Display matching tasks in consistent UI format
- Add input validation and error handling for empty keywords

avinazz3 committed:
Merge branch 'branch-A-Level9': Add find command feature

Add task search functionality allowing users to find tasks by keywords
- Search tasks by description (case-insensitive)
- Format search results in consistent UI style
- Support error handling for invalid inputs

szelingtan commented on src/main/java/babe/Babe.java:
```diff
@@ -0,0 +1,59 @@
+     */
+    public void run() {
+        ui.showWelcome();
+        boolean isExit = false;
```
I like how you made your boolean variable here to sound like booleans

szelingtan commented on src/main/java/babe/parser/DateTimeParser.java:
```diff
@@ -0,0 +1,53 @@
+    private static final DateTimeFormatter STORAGE_FORMATTER =
+            DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+
+    /**
+     * Parses a date-time string into LocalDateTime.
+     * Accepts multiple formats:
+     * - yyyy-MM-dd HHmm (e.g., "2024-02-15 1430")
+     * - d/M/yyyy HHmm (e.g., "15/2/2024 1430")
+     * - d-M-yyyy HHmm (e.g., "15-2-2024 1430")
+     */
```
Perhaps @param dateTimeStr and @throws BabeException may be added with short explanations?

szelingtan commented on src/main/java/babe/task/TaskList.java:
```diff
@@ -0,0 +1,123 @@
+import babe.storage.Storage;
+
+public class TaskList {
+    private ArrayList<Task> tasks;
```
I like that you used the plural form on tasks, which represents a collection of tasks

szelingtan commented on src/main/java/babe/task/TaskList.java:
```diff
@@ -0,0 +1,123 @@
+        }
+    }
+
+    private void saveTaskList() throws BabeException {
```
Perhaps JavaDoc comments can also be given for this method?

szelingtan commented on src/main/java/babe/storage/Storage.java:
```diff
@@ -0,0 +1,177 @@
+
+import babe.task.*;
+import babe.exception.BabeException;
+import java.io.*;
```
Should imported classes be listed explicitly?

szelingtan reviewed (commented):
Overall, I found your code easy to read and generally well documented. I noted that you named your variables and methods well. For imports, I noted that you used the wildcard import instead of listing imported classes explicitly for some files.

avinazz3 committed:
Add JavaFX GUI interface

- Create Launcher class to handle JavaFX initialization
- Implement main chat window with message display area
- Add input field for user commands
- Maintain CLI-style interaction with bye command functionality
- Convert chat display from text-based to graphical interface
- Update documentation for GUI usage

This commit transitions the application from a pure CLI to a
GUI while maintaining the command-line interaction style.
The JavaFX implementation provides a more modern interface
while preserving core chatbot functionality.

avinazz3 committed:
feat: Add assertions to DateTimeParser for input validation and error handling

- Added assertions to ensure input date-time strings are not null or empty.
- Added assertions to validate parsed LocalDateTime objects are not null.
- Added assertions to ensure formatted date-time strings are not empty.
- Improved robustness by documenting assumptions and catching potential issues early.

These changes enhance the reliability of the DateTimeParser class by enforcing
critical assumptions during parsing and formatting operations. Assertions are
used to validate inputs and outputs, ensuring the class behaves as expected
during normal execution.

avinazz3 committed:
refactor: Improve code quality and readability of Storage class

- Extracted logic for validating and creating tasks into separate methods to reduce nesting and improve readability.
- Added validation methods (`validateTaskParts`, `validateDeadlineParts`, `validateEventParts`) to handle errors explicitly.
- Renamed variables and methods for clarity and consistency.
- Applied SLAP (Single Level of Abstraction Principle) to ensure each method focuses on a single level of abstraction.
- Improved error handling and validation to make the code more robust.
- Simplified the `formatTask` method for better readability.
- Added comments to explain the purpose of each method.

These changes enhance the maintainability, readability, and reliability of the Storage class while adhering to coding best practices.

avinazz3 committed:
Merge pull request #1 from avinazz3/branch-A-CodeQuality

Refactor: Improve code quality and readability

avinazz3 committed:
feat: Add assertions to TaskList class for runtime verification

- Add defensive programming checks throughout TaskList to catch bugs early
- Verify critical invariants like null checks and list size changes
- Ensure consistent state after task modifications
- Validate method preconditions and postconditions

avinazz3 committed:
feat: Add javadoc comments to TaskList

avinazz3 committed:
Merge pull request #2 from avinazz3/branch-A-Assertions

Add runtime verification to TaskList implementation through assertions

avinazz3 committed:
feat: Add priority levels to tasks

Implement task prioritization system with three levels (High, Medium, Low):
- Add Priority enum to Task class with levels 1-3
- Update Parser to handle priority input via '/p' flag
- Add SetPriorityCommand for modifying existing task priorities
- Modify Storage to persist priority data
- Update Todo, Deadline, and Event classes with priority support

Example usage:
- Create task with priority: todo Read book /p 1
- Update priority: priority 1 2
- Priority appears in display: [X][P1] Read book

Breaking changes:
- Storage format updated to include priority field
- Existing task constructors require priority parameter
- Task string representation now includes priority level

avinazz3 committed:
Merge pull request #3 from avinazz3/branch-C-Priority

feat: Add priority levels to tasks

avinazz3 committed:
Improve GUI appearance and fix JavaFX loading issues

- Fix architecture compatibility issues by updating JavaFX dependencies for Apple Silicon (arm64)
- Resolve FXML loading errors in MainWindow and DialogBox
- Fix duplicate children issue in DialogBox component
- Improve message styling with proper alignment and colors
- Reduce avatar size to optimize space usage
- Add proper padding and spacing for better visual layout
- Update build configuration for Apple Silicon support

Technical improvements:
- Remove fx:root from MainWindow.fxml and use proper controller pattern
- Clear children before adding new ones in DialogBox to prevent duplicates
- Add proper anchoring of components in MainWindow
- Update JavaFX version to match JDK version

avinazz3 committed:
Merge pull request #4 from avinazz3/branch-A-BetterGui

Improve GUI appearance and fix JavaFX loading issues

avinazz3 committed:
fix: Remove redundant exception handling in priority parser

NumberFormatException is a subclass of IllegalArgumentException, so catching both was redundant. Now only catching the parent exception to handle both cases.

avinazz3 committed:
docs: update README with comprehensive user guide for Babe chatbot

- Add detailed command syntax and examples
- Include task types and priority levels
- Document error handling scenarios
- Add chat window features and interface details
- Include additional features overview

avinazz3 committed:
docs: update README with comprehensive user guide for Babe chatbot

avinazz3 committed:
docs: update README with comprehensive user guide for Babe chatbot

avinazz3 committed:
docs: update README with comprehensive user guide for Babe chatbot

avinazz3 committed:
Add cross platform dependencies into build.gradle

avinazz3 committed:
Add cross platform dependencies into build.gradle

avinazz3 committed:
Add cross platform dependencies into build.gradle
